e2e/deployer.go

```diff
@@ -0,0 +1,54 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// deployer interface modify the templates on demand and also help
+// to create or delete templates.
+type deployer interface {
+	replaceNamespaceInTemplate(string) (string, error)
+	runAction(string, string, string) error
+	runActionRookDelete(string, string, string) error
+}
+
+type deploy struct{}
+
+func (d deploy) replaceNamespaceInTemplate(filePath string) (string, error) {
+	read, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.ReplaceAll(string(read), "namespace: default", fmt.Sprintf("namespace: %s", cephCSINamespace)), nil
+}
+
+func (d deploy) runAction(template, action, ns string) error {
+	data, err := d.replaceNamespaceInTemplate(template)
+	if err != nil {
+		return fmt.Errorf("failed to read content from %s with error %w", template, err)
+	}
+	_, err = framework.RunKubectlInput(cephCSINamespace, data, ns, action, "-f", "-")
+	if err != nil {
+		return fmt.Errorf("failed to %s %q with error %w", action, template, err)
+	}
+
+	return err
+}
+
+func (d deploy) runActionRookDelete(template, action, ns string) error {
```

Comment:
the function name is not generic. we are not doing any specific operation related to rook here.

Reply:
@Madhu-1 any suggestions here ?

Reply:
why not make use of the above runAction method?

Reply:
@Madhu-1 thats why I said above, this method was a placeholder for rook actions as we are seperetely manipulating rook deployment before we start and it also matches with current implementation and give some clarity while invoking the method that, we are doing it on `rook` deployment

Reply:
For me, this doesn't look to be rook specific because we are running in  `cephCSINamespace` with `--ignore-not-found=true` instead of `runActionRookDelete` this will be a wrapper to `runAction` something like`runActionDeleteIgnoreNotFound`

Reply:
@Madhu-1 now you are talking about the real body of this method or what it do , where I was mentioning about from caller's stand point. One thing to note here is our deployment methods/functions are keep doing 'almost' same actions with some changes here and there, generalizing those can be done in a different way or as a follow up.  The initial attempt here is to get the things seggregated to common methods and make use of it. Optimization/refactoring of the real actions inside functions are next items in the list. I have also renamed the method `runActionRookDelete` to `runRookAction` and also marked the todo for actions/wrappers on existing things in the tracker issue.  Let me know if thats enough?

Reply:
still want to make a couple of points here. lets hear from others also

* if we are sure that its a RookAction why to make use of `cephCSINamespace` 
* and we are making use of `--ignore-not-found` it's only specific to delete operations, not for other operations?. please correct me if am wrong.

Reply:
@Madhu-1 the thought here is simple, if we think we dont need to keep room for any other `rook` operations apart from `delete` template, we can remove the placeholder. The idea was that, keep a rook specific function which can be enhanced based on requirement in future.. so, getting rid of `rook` specific method is what you want ?

Reply:
@Madhu-1 any other comments on existing version ?